### PR TITLE
[RDBMS] `az postgres flexible-server migration`: Use Cloud supplied URL's rather than hardcoded management URL's

### PR DESCRIPTION
**Pull request by cpj2195**
**Description**
Earlier, The management URLs were hardcoded because of which the CLI was unusable in the national clouds.
With this change, the CLI will work for all clouds when the user selects the cloud.

**Testing Guide**
Tested manually in the local environment.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
RDBMS

**Comment by DaeunYim** [comment]
Just wanted to make sure if this one is tested in the mooncake region?

File: src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_common.py [hunk 1]
```diff
@@ -73,16 +73,15 @@ def migration_create_func(cmd, client, resource_group_name, server_name, propert
...
+    r = send_raw_request(cmd.cli_ctx, "put", cmd.cli_ctx.cloud.endpoints.resource_manager + "subscriptions/{}/resourceGroups/{}/providers/Microsoft.DBforPostgreSQL/flexibleServers/{}/migrations/{}?api-version=2022-05-01-privatepreview".format(subscription_id, resource_group_name, server_name, migration_name), None, None, json_data)
     return r.json()
 
 
 def migration_show_func(cmd, client, resource_group_name, server_name, migration_name):
 
     subscription_id = get_subscription_id(cmd.cli_ctx)
 
-    r = send_raw_request(cmd.cli_ctx, "get", "https://management.azure.com/subscriptions/{}/resourceGroups/{}/providers/Microsoft.DBforPostgreSQL/flexibleServers/{}/migrations/{}?api-version=2022-05-01-privatepreview".format(subscription_id, resource_group_name, server_name, migration_name))
+    r = send_raw_request(cmd.cli_ctx, "get", cmd.cli_ctx.cloud.endpoints.resource_manager + "subscriptions/{}/resourceGroups/{}/providers/Microsoft.DBforPostgreSQL/flexibleServers/{}/migrations/{}?api-version=2022-05-01-privatepreview".format(subscription_id, resource_group_name, server_name, migration_name))
```
Comment (alanenriqueo) [inline src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_common.py:84]: `cmd.cli_ctx.cloud.endpoints.resource_manager` may or may not contain trailing `/`. An easy way to fix this is just remove `https://management.azure.com`, function `send_raw_request` will take care of appending the current cloud resource manager endpoint correctly.

**Review by alanenriqueo — CHANGES_REQUESTED** [review]

**Comment by cpj2195** [comment]
> Just wanted to make sure if this one is tested in the mooncake region?

@DaeunYim , the FMS based migrations are not available in Mooncake region right now. Since we have removed the hardcoded management URL's, should it work out of the box?

**Review by alanenriqueo — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_common.py [hunk 2]
```diff
@@ -139,15 +138,16 @@ def migration_update_func(cmd, client, resource_group_name, server_name, migrati
     if operationSpecified is False:
         raise RequiredArgumentMissingError("Incorrect Usage: At least one update operation needs to be specified.")
 
-    r = send_raw_request(cmd.cli_ctx, "patch", "https://management.azure.com/subscriptions/{}/resourceGroups/{}/providers/Microsoft.DBforPostgreSQL/flexibleServers/{}/migrations/{}?api-version=2022-05-01-privatepreview".format(subscription_id, resource_group_name, server_name, migration_name), None, None, properties)
+    server_endpoint = cmd.cli_ctx.cloud.suffixes.postgresql_server_endpoint
```
Comment (alanenriqueo) [inline src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_common.py:141]: Remove this unused variable so style check passes.

**Review by alanenriqueo — APPROVED** [review]

**Review by evelyn-ys — APPROVED** [review]

**Pull request merged**